api/src/main/java/org/apache/iceberg/SnapshotRef.java

```diff
@@ -86,7 +86,7 @@ public boolean equals(Object other) {
     }
 
     SnapshotRef ref = (SnapshotRef) other;
-    return Objects.equals(ref.snapshotId(), snapshotId) &&
+    return ref.snapshotId() == snapshotId &&
```

Comment:
I think it's fine if we disable the static analysis rule for this, just because I like it when all the comparisons look the same but if we want to follow this rule that's also ok.

Reply:
revert